File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/tasks/geodatabase/GenerateGeodatabaseJobNativeObject.java
package com.valentingrigorean.arcgis_maps_flutter.tasks.geodatabase;

import com.esri.arcgisruntime.tasks.geodatabase.GenerateGeodatabaseJob;
import com.valentingrigorean.arcgis_maps_flutter.concurrent.JobNativeHandler;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.ArcgisNativeObjectsController;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.BaseNativeObject;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.NativeHandler;
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.NativeMessageSink;

public class GenerateGeodatabaseJobNativeObject extends BaseNativeObject<GenerateGeodatabaseJob> {

    public GenerateGeodatabaseJobNativeObject(String objectId, GenerateGeodatabaseJob job, NativeMessageSink messageSink) {
        super(objectId, job, new NativeHandler[]{
                new JobNativeHandler(job, JobNativeHandler.JobType.GENERATE_GEODATABASE)
        });
        setMessageSink(messageSink);
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/tasks/geodatabase/GenerateGeodatabaseJobNativeObject.kt
package com.valentingrigorean.arcgis_maps_flutter.tasks.geodatabase

import com.arcgismaps.tasks.geodatabase.GenerateGeodatabaseJob
import com.valentingrigorean.arcgis_maps_flutter.concurrent.JobNativeHandler
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.BaseNativeObject
import com.valentingrigorean.arcgis_maps_flutter.flutterobject.NativeMessageSink

class GenerateGeodatabaseJobNativeObject(
    objectId: String,
    job: GenerateGeodatabaseJob,
    messageSink: NativeMessageSink
) : BaseNativeObject<GenerateGeodatabaseJob>(
    objectId, job, arrayOf(
        JobNativeHandler(job, JobNativeHandler.JobType.GENERATE_GEODATABASE)
    )
) {
    init {
        setMessageSink(messageSink)
    }
}